lib/internal/repl.js

```diff
@@ -92,7 +92,15 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
 
   function oninit(err, hnd) {
     if (err) {
-      return ready(err);
+      // Cannot open history file.
+      // Don't crash, just don't persist history.
+      debug('oninit: %s', err.message);
+      repl._writeToOutput('\nError: Could not open history file.\n' +
+                          'REPL session history will not be persisted.\n');
+      repl._refreshLine();
+      repl._historyPrev = _replHistoryMessage;
+      repl.resume();
```

Comment:
Yea, it is paused explicitly at the beginning of setupHistory. If we don't resume, then the process will exit.